### PR TITLE
Kubeadm e2e

### LINKED ISSUE #456: Create `[Feature:kubeadm]` e2e tests
...under `test/e2e_kubeadm/` in the core kubernetes repo.
e2e tests here should for instance make sure that:
 - At least one node is tainted/labelled with `node-role.kubernetes.io/master=""`
 - `cluster-info` is exposed in a correct manner, and the right `Role` and `RoleBinding` exist
 - The right Node Bootstrap Token RBAC rules exist, the SAR approver is enabled, etc.
 - The `kubeadm-config` ConfigMap exists.
 - API Aggregation features are enabled
 - What else?

@timothysc @pipejakob Ideas?

### PR DESCRIPTION
**Pull request by liztio**
**What this PR does / why we need it**:

Provides in-tree E2E tests for the Kubeadm subproject

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes [kubeadm#456](https://github.com/kubernetes/kubeadm/issues/456)

**Special notes for your reviewer**:

The weird way tests are executed mirrors `e2e_node`. A future pull request will add a frontend for these tests to kubetest, which will abstract away much of this detail.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e_kubeadm/runner/local/run_local.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	args := append([]string{"build"}, targets...)
+
+	return execCommand("bazel", args...)
+}
+
+var ginkgoFlags = flag.String("ginkgo-flags", "", "Space-separated list of arguments to pass to Ginkgo test runner.")
+var testFlags = flag.String("test-flags", "", "Space-separated list of arguments to pass to node e2e test.")
+var build = flag.Bool("build", false, "use Bazel to build binaries before testing")
+
+func main() {
```
Comment (timothysc) [inline test/e2e_kubeadm/runner/local/run_local.go]: why is this here?

File: test/e2e_node/builder/build.go [hunk 2]
```diff
@@ -38,18 +38,18 @@ var buildTargets = []string{
 }
 
 func BuildGo() error {
-	glog.Infof("Building k8s binaries...")
-	k8sRoot, err := GetK8sRootDir()
+	glog.Info("Building k8s binaries...")
```
Comment (timothysc) [inline test/e2e_node/builder/build.go]: This seems like an orthogonal set of changes which should be their own PR.

**Review by timothysc — CHANGES_REQUESTED** [review]
IMO this PR should be split into multiple.

**Comment by liztio** [comment]
@timothysc the utils PR has been merged, this PR now only the E2E-specific changes

File: test/e2e_kubeadm/runner/local/run_local.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/test/utils"
+)
+
+func bazelBuild() error {
```
Comment (timothysc) [inline test/e2e_kubeadm/runner/local/run_local.go]: I'm still confused why this is needed.

**Review by timothysc — COMMENTED** [review]
Why do you need the bazel shim, does the bazel build not properly create the test artifact?

File: test/e2e_kubeadm/runner/local/run_local.go (hunk 3 above)
Comment (liztio) [reply test/e2e_kubeadm/runner/local/run_local.go]: it's just how the e2e_node tests work. The local runner is responsible for compiling the artefact. I could take it out, but that responsibility has to be somewhere.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

I'll circle back on the build mechanics later.
